src/decode-ipv4.c

```diff
@@ -485,7 +485,7 @@ static int DecodeIPV4Options(Packet *p, uint8_t *pkt, uint16_t len, IPV4Options
     return 0;
 }
 
-static int DecodeIPV4Packet(Packet *p, uint8_t *pkt, uint16_t len)
+static int DecodeIPV4Packet(Packet *p, uint8_t *pkt, uint32_t len)
```

Comment:
so in what condition could an IP packet be bigger than 64k?

Reply:
You are perfectly right, there is no possibility for an IP packet to be bigger than 64k. However, the variable passed is a uint32 (from DecodeEthernet). I modified these two decoders (both IP and TCP) to get to a standard among decoders. Of course, if you prefer to restrict the change only where is needed, I will just remove the layer 3 and layer 4 - related modification.